lib/modules/apostrophe-nuanced-permissions-doc-type-manager/index.js

```diff
@@ -0,0 +1,137 @@
+const _ = require('lodash');
+
+module.exports = {
+
+  improve: 'apostrophe-doc-type-manager',
+
+  afterConstruct: function(self) {
+    // We have to do this late because they are introduced
+    // in subclasses
+    self.composeNuancedPermissions();
+    self.overrideEditControls();
+  },
+
+  construct: function(self, options) {
+
+    self.overrideEditControls = function() {
+      if (self.getEditControls) {
+        const superGetEditControls = self.getEditControls;
+        self.getEditControls = function(req) {
+          const has = self.getNuancedPermission(req);
+          if (!has) {
+            return superGetEditControls(req);
+          }
+          if (!has.update) {
+            return superGetEditControls(req);
+          }
+          return [
+            {
+              type: 'minor',
+              action: 'cancel',
+              label: 'Cancel'
+            }
+          ].concat(has.submit ? [
+            {
+              type: 'minor',
+              label: 'Submit',
+              action: 'workflow-submit'
+            }
```

Comment:
Does this assume `apostrophe-workflow` is enabled in the project? If not, and that module is optional, should be throwing warnings if a user is configuring a selective permission on an action that isn't available in any installed modules of the CMS?